src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -86,36 +86,53 @@ const ioTask = ({ name }) => _.nth(-2, name.split('.'))
 const ioVariable = ({ name }) => _.nth(-1, name.split('.'))
 const ioType = ({ inputType, outputType }) => (inputType || outputType).match(/(.*?)\??$/)[1] // unify, and strip off trailing '?'
 
+
 const WorkflowIOTable = ({ which, inputsOutputs: data, config, errors, onChange, onSetDefaults, onBrowse, suggestions, readOnly }) => {
+  const [sort, setSort] = useState({ field: 'taskVariable', direction: 'asc' })
+
+  const sortedData = _.orderBy([
```

Comment:
One possible simplification, although @panentheos can doubtless think of something better:
```js
const sortedData = _.orderBy([
  'optional',
  o => (sort.field === 'taskVariable' ? ioTask : ioVariable)(o).toLowerCase(),
  o => (sort.field === 'taskVariable' ? ioVariable : ioTask)(o).toLowerCase()
], ['asc', sort.direction, sort.direction], data)
```

Reply:
The current form makes 'optional' the primary sort in all cases, which feel unintuitive to me. Is that what we want? (Also note that if we put it last, it will probably have no effect because the variable names are unique. I'm not sure what the best option is).

Reply:
As for the code, what about something like this?
```js
const taskSort = o => ioTask(o).toLowerCase()
const varSort = o => ioVariable(o).toLowerCase()
const sortedData = _.orderBy(
  sort.field === 'taskVariable' ? ['optional', taskSort, varSort] : ['optional', varSort, taskSort],
  ['asc', sort.direction, sort.direction]
)
```
This communicates that this code has two parts: a couple of iteratees, and a condition that chooses a sequence of sorts using those iteratees.

Reply:
Me and @breilly2 put `optional` first so that we could guarantee that all rows with options were always at the bottom and then sorted within the categories of without optional and with optional (which we understood was ideal behavior)

Reply:
Yeah... other options would be
* add a column for "optional" to allow explicit sorting on that
* remove the code that puts optional inputs below all others at the start (because that state cannot be achieved again after performing some other sort)

Reply:
@breilly2 @panentheos do we want to create a follow-on ticket to discuss this? Is it okay if I merge this as is?

Reply:
@breilly2 @panentheos do we want to create a follow-on ticket to discuss this? Is it okay if I merge this as is?